### ISSUE TITLE
queryContext is missing in type declarations

### ISSUE THREAD

**Issue by MoonCall** [tgriesser/knex#2930]
# Environment

Knex version: 0.15.2

(I think those are not needed information, however, I included)
Database + version: MySQL  5.7.22
OS: OSX 10.12.6

# Bug

I found .queryContext is supported as written in knex documents(https://knexjs.org/#Builder-queryContext), and used that in one of my projects. However, when I applied @types/knex typescript compile error occured because queryContext is missing in type declaration file (and also types in this repository does not include type information about queryContext)

```
error TS2339: Property 'queryContext' does not exist on type 'QueryBuilder'.

47   return client(tableName).queryContext(tableName);
```

I think `queryContext(queryContext: any): QueryBuilder;` needs to be added to QueryBuilder interface.

I found knex recently decided to manage types internally from #2845, so I made issue here first.
If this problem should be issued in DefinitelyTyped also, I will.

**Comment by kibertoad** [comment]
@MoonCall It is correct to report it here; would you consider raising a PR with fix?

**Comment by MoonCall** [comment]
@kibertoad Sure, I'll make a PR soon.

**Comment by kibertoad** [comment]
Thanks!

**Comment by MoonCall** [comment]
@kibertoad 
I made a PR for this issue (#2931).

While I was working on this, I found queryContext is missing not only in QueryBuilder but also in SchemaBuilder, TableBuilder and ColumnBuilder. Of course queryContext is also missing in Raw, but I skipped Raw because its documentation does not list queryContext as a possible method. Finally, in addition to adding `queryContext(context: any): QueryBuilder` to QueryBuilder interface, I also added those to each of mentioned above.

- `queryContext(context: any): SchemaBuilder`
- `queryContext(context: any): TableBuilder`
- `queryContext(context: any): ColumnBuilder`
